treeherder/client/thclient/perfherder.py

```diff
@@ -0,0 +1,105 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from .client import TreeherderClient, TreeherderClientError
+
+
+class TimeInterval(object):
+    DAY = 86400
+    WEEK = 604800
+    TWO_WEEKS = 1209600
+    SIXTY_DAYS = 5184000
+    NINETY_DAYS = 7776000
+
+
+class SignatureList(object):
+
+    def __init__(self, signatures):
+        self.signatures = signatures
+
+    def __len__(self):
+        return len(self.signatures)
+
+    def __getitem__(self, key):
+        return self.signatures.get(key)
+
+    def filter(self, *args):
+        filtered_signatures = {}
+        for (signature, signature_value) in self.signatures.iteritems():
+            skip = False
+            for (key, val) in args:
+                if signature_value.get(key) != val:
+                    skip = True
+                    break
+            if not skip:
+                filtered_signatures[signature] = signature_value
+        return SignatureList(filtered_signatures)
+
+    def get_signature_hashes(self):
+        return self.signatures.keys()
+
+    def get_property_names(self):
+        property_names = set()
+        for signature_value in self.signatures.values():
+            for property_name in signature_value.keys():
+                property_names.add(property_name)
+        return property_names
+
+    def get_property_values(self, property_name):
+        property_values = set()
+        for signature_value in self.signatures.values():
+            if signature_value.get(property_name):
+                property_values.add(signature_value[property_name])
+        return property_values
+
+
+class Series(object):
```

Comment:
This class looks like a smart version of a list, maybe you want to inherit from it?